internal/provider/waypoint/resource_waypoint_action_config.go

```diff
@@ -0,0 +1,696 @@
+						Optional:    true,
+						Attributes: map[string]schema.Attribute{
+							"method": schema.StringAttribute{
+								Description: "The HTTP method to use for the request.",
+								Required:    true,
+							},
+							"headers": schema.MapAttribute{
+								Description: "Key value headers to send with the request.",
+								Optional:    true,
+								ElementType: types.StringType,
+							},
+							"url": schema.StringAttribute{
+								Description: "The full URL this request should make when invoked.",
+								Optional:    true,
+							},
+							"body": schema.StringAttribute{
+								Description: "The body to be submitted with the request.",
+								Optional:    true,
+							},
+						},
+					},
+					/*"github": schema.ListNestedAttribute{
+						Description: "GitHub mode is configured to do various operations on GitHub Repositories.",
+						Optional:    true,
+					},
+					"agent": schema.ListNestedAttribute{
+						Optional: true,
+					},*/
+				},
+			},
+		},
+	}
+}
+
+func (r *ActionConfigResource) Configure(ctx context.Context, req resource.ConfigureRequest, resp *resource.ConfigureResponse) {
+	// Prevent panic if the provider has not been configured.
+	if req.ProviderData == nil {
+		return
+	}
+
+	client, ok := req.ProviderData.(*clients.Client)
+	if !ok {
+		resp.Diagnostics.AddError(
+			"Unexpected Resource Configure Type",
+			fmt.Sprintf("Expected *http.Client, got: %T. Please report this issue to the provider developers.", req.ProviderData),
+		)
+
+		return
+	}
+
+	r.client = client
+}
+
+func (r *ActionConfigResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	var plan *ActionConfigResourceModel
+
+	// Read Terraform plan data into the model
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &plan)...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	projectID := r.client.Config.ProjectID
+	if !plan.ProjectID.IsUnknown() {
+		projectID = plan.ProjectID.ValueString()
+	}
+
+	orgID := r.client.Config.OrganizationID
+	loc := &sharedmodels.HashicorpCloudLocationLocation{
+		OrganizationID: orgID,
+		ProjectID:      projectID,
+	}
+
+	client := r.client
+	ns, err := getNamespaceByLocation(ctx, client, loc)
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"error getting namespace by location",
+			err.Error(),
+		)
+		return
+	}
+
+	modelBody := &waypoint_models.HashicorpCloudWaypointWaypointServiceCreateActionConfigBody{
+		ActionConfig: &waypoint_models.HashicorpCloudWaypointActionConfig{
+			Request: &waypoint_models.HashicorpCloudWaypointActionConfigRequest{},
+		},
+	}
+
+	if !plan.Name.IsUnknown() {
+		modelBody.ActionConfig.Name = plan.Name.ValueString()
+	}
+
+	if !plan.Description.IsUnknown() {
+		modelBody.ActionConfig.Description = plan.Description.ValueString()
+	}
+
+	if !plan.ActionURL.IsUnknown() {
+		modelBody.ActionConfig.ActionURL = plan.ActionURL.ValueString()
+	}
+
+	var diags diag.Diagnostics
```

Comment:
This is actually necessary, when trying to use the assignment `:=` in those locations the compiler complains so without the initial declaration it will not work.

Reply:
This file has changed since I commented, it's now `resource_waypoint_action.go`, but the comment remains:

on line 244 and 301 you can use `:=` and remove the `var diag` declaration since there's no need to declare it in the larger scope.

Reply:
I apologize, I should have first asked if you could share the compiler warning, could you paste that here if it's still complaining? Maybe I'm missing something!